File: HydrusSharp/Models/Client/HashedJsonDump.cs
using Newtonsoft.Json;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace HydrusSharp.Models.Client
{
    [Table("json_dumps_hashed")]
    public class HashedJsonDump
    {
        [Column("hash")]
        public byte[] Hash { get; set; }

        [Column("dump_type")]
        public int DumpType { get; set; }

        [Column("version")]
        public int Version { get; set; }

        [Key]
        [Column("dump")]
        public string Dump { get; set; }

        [NotMapped]
        public string HashedString => BitConverter.ToString(Hash).Replace("-", "").ToLowerInvariant();
    }
}

File: HydrusSharp/Core/Scripts/ViewModels/Page/HashedJsonDump.ts
class HashedJsonDump {

    public dumpType: ko.Observable<number>;

    public version: ko.Observable<number>;

    public dump: ko.Observable<string>;

    public hash: ko.Observable<string>;

    public managementController: ko.Observable<ManagementController>;

    public constructor(view: IHashedJsonDump) {

        this.dumpType = ko.observable(view.DumpType);
        this.version = ko.observable(view.Version);
        this.dump = ko.observable(view.Dump);
        this.hash = ko.observable(view.HashedString);

        this.managementController = ko.observable(new ManagementController(JSON.parse(view.Dump)[0]));
    }
}